### ISSUE TITLE
CDS Header Action Click property not propagating 

### ISSUE THREAD

**Issue by ParkerMIBM** [carbon-design-system/carbon-components-angular#2698]
## Detailed description

> Migrating from carbon-component-angular@4.59 to 5.14, both cds-header-action and ibm-header-action are no longer calling the function I specified in the click property. I see the "active" and "selected" properties being updated correctly, but I am unsure why the (click)="refreshDataOnPage($event)" that has long existed on this element would stop being called.

> Is this a feature request (new component, new icon), a bug, or a general issue?
This is a bug

> Is this issue related to a specific component?
This is related to the UIShell - specifically the cds-header-action component. 

> What did you expect to happen? What happened instead? What would you like to see changed?
The v4 to v5 migration document linked in this repository had me expecting to change from description to action. It said nothing about how events would propagate. I would have expected that my (click) property would have continued to work as normal.

> What browser are you working in?
Firefox, chrome, safari

> What version of the Carbon Design System are you using?
    "@carbon/icons": "11.26.0",
    "@carbon/styles": "1.37.0",
    "@carbon/themes": "11.24.0",
    "carbon-components-angular": "5.14.10",
    "@angular/core": "14.3.0",

## Steps to reproduce the issue
I tried to recreate on the code sandbox but it was very down level and updating it to try to recreate my environment was starting to take too long through that interface.
Basically
1. Have a UIShellHeader bar with a <ibm-header-action> component, what was displayed was an icon like in the story.
2. create a function within the component "public myOnClickFunction( event ) { console.log("we made it!"); }"
3. add a click property to propagate an onclick <ibm-header-action (click)="myOnClickFunction($event)">
4. Load page
5. Click button
6. notice in console log no "we made it!" message

## Additional information

* I am not getting any console errors
* The angular inspection tools on chrome do not show the click property on the element

**Comment by Akshat55** [comment]
Please use the `selected` and an `activeChange` event.
